File: solutions/_209_minimum_size_subarray_sum/solution.java

package _209_minimum_size_subarray_sum;

class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int min = Integer.MAX_VALUE;
        int sum = 0;
        int L = 0;

        for (int R = 0; R < nums.length; R++) {
            sum += nums[R];
            while (sum >= target) {
                min = Math.min(min, R - L + 1);
                sum -= nums[L];
                L++;
            }
        }

        return min == Integer.MAX_VALUE ? 0 : min;
    }
}

/*
 * https://leetcode.com/problems/minimum-size-subarray-sum/
 */

File: solutions/_209_minimum_size_subarray_sum/solution.py
from typing import List


class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        res = float("inf")
        L = 0
        total = 0

        for R in range(len(nums)):
            total += nums[R]
            while total >= target:
                res = min(res, R - L + 1)
                total -= nums[L]
                L += 1

        return 0 if res == float("inf") else res


# https://leetcode.com/problems/minimum-size-subarray-sum/
